client/src/app/appCtrl.js

```diff
@@ -181,8 +197,13 @@ angular.module('app').controller('appCtrl', ['$scope', '$http', '$timeout', 'app
     firstDataLoaded = false;
   };
 
-  var downloadFile = function(url) {
+  // function to "download" a file, 
+  // with param. 'mode': "download","streamLocal","streamRemote"
+  var downloadFile = function(url, mode) {
```

Comment:
I made a cleanup of the 3 functions: `downloadFile(), streamRemote(), streamLocal()` and unified them to this parametrized method